src/prog_models/prognostics_model.py

```diff
@@ -1085,4 +1087,344 @@ def optimization_fcn(params):
-        self.parameters['process_noise'] = p_noise                
+        self.parameters['process_noise'] = p_noise   
+
+    def generate_surrogate_dmd(self, load_functions, first_output = None, threshold_keys = None, **kwargs):
+        """
+        Generate a surrogate model to approximate the higher-fidelity model 
+
+        Parameters
+        ----------
+        load_functions : list of callable functions
+            Each index is a callable loading function of (t) -> z used to predict future loading (output) at a given time (t)
+
+        Keyword Arguments
+        -----------------
+        Includes all keyword arguments from simulate_to_threshold (except save_pts), and the following additional keywords: 
+
+        dt : Number or function, optional
+            Same as in simulate_to_threshold; for DMD, this value is the time step of the training data\n
+        save_freq : Number, optional
+            Same as in simulate_to_threshold; for DMD, this value is the time step with which the surrogate model is generated  \n
+        data_len: int, optional
+            Value between 0 and 1 that determines fraction of data resulting from simulate_to_threshold that is used to train DMD surrogate model
+            e.g. if data_len = 0.7 and the simulated data spans from t=0 to t=100, the surrogate model is trained on the data from t=0 to t=70 \n        
+        states_dmd: list, optional
+            List of state keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n
+        inputs_dmd: list, optional
+            List of input keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n
+        outputs_dmd: list, optional
+            List of output keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n 
+        events_dmd: list, optional
+            List of event state keys (must match those defined in the PrognosticsModel) to be included in the surrogate model generation \n           
+        
+        Returns
+        -------
+        SurrogateModel(): class
+            Instance of SurrogateModel class
+
+        Example
+        -------
+        See examples/generate_surrogate
+        
+        """
+
+        # Generate Data to train surrogate model: 
+
+        # Configure
+        config = { # Defaults
+            't0': 0.0,
+            'dt': 1.0, 
+            'save_freq': 1.0, 
+            'horizon': 1e100, # Default horizon (in s), essentially inf
+            'print': False,
+            'progress': False,
+            'data_len': 1,
+            'states_dmd': self.states,
+            'inputs_dmd': self.inputs,
+            'outputs_dmd': self.outputs,
+            'events_dmd': self.events
+        }
+        config.update(kwargs)
+
+        # List of user-define values to include in surrogate model: 
+        states_dmd = deepcopy(self.states)
```

Comment:
I came across an issue where if the user inputs the states in a different order than what's in self.states, there was order issues later on within SurrogateModel because the matrix was set up with the states in the wrong order (I can't quite remember the details, but that was the gist - I think the DMD matrix is in the order of self.states but SurrogateModel tries to iterate with a state vector of the order in config['states_dmd']). This was the fix I was able to come up with, but we can look at this together if you think there's a better way